File: Roadmap/18 - CONJUNTOS/python/Sac-Corts.py
my_list = [1, 2, 3, 4, 5]
my_list.append(6)
print(my_list)

my_list.insert(0, 0)
print(my_list)

my_list.extend([7, 8, 9])
print(my_list)

my_list[4:4] = [22, 23]
print(my_list)

del my_list[4]
del my_list[4]
print(my_list)

my_list[9] = 10
print(my_list)

element = 10
in_list = element in my_list
print(in_list)

my_list.clear()
print(my_list)

### Ejercicio Extra ###

set_a = {1, 2, 3, 4, 5}
set_b = {4, 5, 6, 7, 8}
print("Conjunto a:", set_a)
print("Conjunto b:", set_b)

# Unión
union = set_a | set_b
print("Unión:", union)

# Intersección
intersection = set_a & set_b
print("Intersección:", intersection)

# Diferencia
difference = set_a - set_b
print("Diferencia:", difference)

# Diferencia Simétrica
symmetrical_difference = set_a ^ set_b
print("Diferencia Simétrica:", symmetrical_difference)

File: Roadmap/18 - CONJUNTOS/typescript/Sac-Corts.ts
let set: number[] = [];
set.push(10);
set.unshift(1);
set.push(11, 12);
set.splice(1, 0, 2, 3);
set.splice(2, 1);
set[2] = 5;
let elementExist: boolean = set.includes(3);
console.log(elementExist);
console.log(set);
set = [];
console.log(set);

// ** Extra Exercise ** //
const setA: Set<number> = new Set([1, 2, 3]);
const setB: Set<number> = new Set([3, 4, 5]);

// Union
const union: Set<number> = new Set([...setA, ...setB]);
console.log(union);

// Intersection
const intersection: Set<number> = new Set([...setA].filter(x => setB.has(x)));
console.log(intersection);

// Difference
const difference: Set<number> = new Set([...setA].filter(x => !setB.has(x)));
console.log(difference);

// Symmetric Difference
const symmetricDifference: Set<number> = new Set([
    ...[...setA].filter(x => !setB.has(x)), 
    ...[...setB].filter(x => !setA.has(x))
]);
console.log(symmetricDifference);
